src/api/hosts.ts

```diff
@@ -0,0 +1,7 @@
+import axios, { AxiosPromise } from 'axios';
+import { Host } from '../models/hosts';
+import { ApiResponse } from './index';
+
+type HostListApiReponse = AxiosPromise<ApiResponse<Host[]>>;
```

Comment:
Nit: I am inclined to rename this to 'GetHostsApiResponse'

Reply:
Yep, works for me.